### PR TITLE
feat(gatsby): let serve use getServerData headers

### PR DESCRIPTION
**Pull request by pieh**
## Description

This exposes custom headers set by `getServerData` to engine integration and actually allows setting those reponse headers in `gatsby serve`

[ch37799]

### CODE REVIEW

File: packages/gatsby/src/utils/page-ssr-module/entry.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export interface ISSRData {
   page: IGatsbyPage
   templateDetails: ITemplateDetails
   potentialPagePath: string
+  serverDataHeaders?: Record<string, string | number | Array<string>>
```
Comment (wardpeet) [inline packages/gatsby/src/utils/page-ssr-module/entry.ts:31]: This should always be a string. Headers can only hold string values, if they hold anything else it get converted
```suggestion
  serverDataHeaders?: Record<string, string>
```

File: packages/gatsby/src/utils/get-server-data.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { match } from "@gatsbyjs/reach-router/lib/utils"
 
 export interface IServerData {
   headers?: Record<string, string>
-  props?: Record<string, unknown>
+  props?: Record<string, string | number | Array<string>>
```
Comment (wardpeet) [inline packages/gatsby/src/utils/get-server-data.ts:8]: I would keep this unknown

**Review by wardpeet — COMMENTED** [review]
A few comments about types

File: packages/gatsby/src/utils/page-ssr-module/entry.ts (hunk 1 above)
Comment (pieh) [reply packages/gatsby/src/utils/page-ssr-module/entry.ts:31]: I followed https://microsoft.github.io/PowerBI-JavaScript/classes/_node_modules__types_node_http_d_._http_.outgoingmessage.html#setheader which is signature for function I later set headers on `res` in `gatsby serve`. I'm fine with the change - just checking if you are sure if it's currect one?

File: packages/gatsby/src/utils/get-server-data.ts (hunk 2 above)
Comment (pieh) [reply packages/gatsby/src/utils/get-server-data.ts:8]: Right, also this wasn't place I want to change (I wanted to change `headers` not `props`), but given previous comment I'll probably just revert this

Comment (wardpeet) [reply packages/gatsby/src/utils/get-server-data.ts:8]: I'm okay with the above. Node will take care of it

**Review by wardpeet — APPROVED** [review]
LGTM

**Pull request merged**